ffcx/ir/analysis/modified_terminals.py

```diff
@@ -187,6 +192,10 @@ def analyse_modified_terminal(expr):
     reference_value = None
     restriction = None
     averaged = None
+    if cell is not None and expr.ufl_domain() is not None:
+        codim = cell.topological_dimension() - expr.ufl_domain().topological_dimension()
```

Comment:
```suggestion
        codim = cell.topological_dimension() - expr.ufl_domain().topological_dimension()
        assert codim >= 0
```

Reply:
I agree with Jørgen's comment (we need to make sure that `codim >= 0`), but maybe even better to have an explicit error message when it is not the case